Nightly wheels are out of date ?

Astropy's devdeps tests are currently crashing at startup

For instance, see https://github.com/astropy/astropy/actions/runs/8119881813/job/22196363368?pr=16087

```python-traceback
...
packages/erfa/__init__.py", line 5, in <module>
    from .version import version as __version__  # noqa
  File "/home/runner/work/astropy/astropy/.tox/py312-test-devdeps/lib/python3.12/site-packages/erfa/version.py", line 25, in <module>
    from . import ufunc
RuntimeError: module compiled against ABI version 0x1000009 but this version of numpy is 0x2000000
```

From what I can see, my guess is that the latest nightlies available at https://pypi.anaconda.org/liberfa/simple are out of date and not compatible with the numpy's latest  nightly builds. To be more exact, at the time of writing, numpy nightlies were last uploaded 17 hours ago (see https://github.com/numpy/numpy/actions/workflows/wheels.yml), but pyerfa's last successful upload is from 29 hours ago (see https://github.com/liberfa/pyerfa/actions).
This would normally not be a problem since new nightlies are supposed to be uploaded well, nightly. However it looks like the automation is currently flaky and failing more often than not (see https://github.com/liberfa/pyerfa/actions). Specifically, Windows builds are failing, and it seems that nothing gets uploaded unless every build succeeds. I'd suggest changing this last rule as a quick way to mitigate the problem (I suspect most nightly users only care about Linux wheels anyway). I'm happy to open a PR on Monday if that is still relevant (there are 2 other scheduled uploads between now and then, a single succesful one would resolve the issue downstream, as far as I can tell).

I have not researched the reason why windows builds are flaky (or non deterministic, for what it's worth), but I'm happy to give a hand in solving the root problem if I can. Let me know !

So the error we get (twice) is:
```
  erfa\ufunc.c(9714): error C2057: expected constant expression
  erfa\ufunc.c(9714): error C2466: cannot allocate an array of constant size 0
  erfa\ufunc.c(9714): error C2133: 'types_array': unknown size
```
which point to this line:
```c
    int types_array[NPY_MAXARGS];
```
The issue here is that `NPY_MAXARGS` is now runtime dependent (32 with numpy 1.x, 64 with numpy 2) (see https://numpy.org/devdocs/reference/c-api/array.html#c.NPY_MAXARGS), which the compiler interprets as "this could be 0", and allocating a 0-size array isn't allowed, hence the compilation error.

I further note that the migration guide explicitly discourages use of this constant
https://numpy.org/devdocs/numpy_2_0_migration_guide.html#increased-maximum-number-of-dimensions
What's not yet clear to me is wether we can just drop it and why it's only showing up on Windows, but we're getting close !

--- a/erfa/helpers.py
+++ b/erfa/helpers.py
@@ -14,6 +14,8 @@ from .core import ErfaWarning
 from .ufunc import get_leap_seconds, set_leap_seconds, dt_eraLEAPSECOND
 
 
+NUMPY_LT_2_0 = np.__version__.startswith("1.")
+
 _NotFound = object()
 
 
@@ -269,8 +271,8 @@ class leap_seconds:
         if hasattr(table, '__array__'):
             table = table.__array__()[list(dt_eraLEAPSECOND.names)]
 
-        table = np.array(table, dtype=dt_eraLEAPSECOND, copy=False,
-                         ndmin=1)
+        table = np.array(table, dtype=dt_eraLEAPSECOND, ndmin=1,
+                         copy=False if NUMPY_LT_2_0 else None)
 
         # Simple sanity checks.
         if table.ndim > 1:

--- a/erfa/ufunc.c.templ
+++ b/erfa/ufunc.c.templ
@@ -572,7 +572,9 @@ static int ErfaUFuncTypeResolver(PyUFuncObject *ufunc,
 {
     int *types;
     PyArray_Descr **dtypes;
-    int types_array[NPY_MAXARGS];
+
+    /* This array needs to be at least as long as nint+nout. In practice, 32 suffices */
+    int types_array[32];
 
     if (ufunc->userloops) {
         Py_ssize_t unused_pos = 0;
@@ -752,7 +754,10 @@ PyMODINIT_FUNC PyInit_ufunc(void)
     PyArray_Descr *dt_ymdf = NULL, *dt_hmsf = NULL, *dt_dmsf = NULL;
     PyArray_Descr *dt_sign = NULL, *dt_type = NULL;
     PyArray_Descr *dt_eraASTROM = NULL, *dt_eraLDBODY = NULL;
-    PyArray_Descr *dtypes[NPY_MAXARGS];
+
+    /* This array needs to be at least as long as nint+nout. In practice, 32 suffices */
+    PyArray_Descr *dtypes[32];
+
     /* ufuncs and their definitions */
     int status;
     PyUFuncObject *ufunc;

--- a/tox.ini
+++ b/tox.ini
@@ -45,6 +45,10 @@ deps =
 extras =
     test
 
+install_command =
+    !devdeps: python -I -m pip install
+    devdeps: python -I -m pip install -v --pre
+
 commands =
     pip freeze
     python -c 'import erfa'  # To give useful error message if liberfa is too old.

